### DISCUSSION TITLE
Implementing into more complex Jekyll themes

### DISCUSSION THREAD

**Discussion by ayan20985** [evanbaldonado/github-pages-jekyll-password-protection#2] [General]
Hello!
This is quite awesome work and I in the middle of trying to integrate this into a Jekyll theme called [Basically Basic](https://github.com/mmistakes/jekyll-theme-basically-basic) but I've been having a lot of trouble trying to get this to password protect a page like index.html or about.md.

Is there any chance I could be guided through the process of integrating your code into a more complex theme @evanbaldonado?

My code is available over [here](https://github.com/420Ayan420/420ayan420.github.io).

Many thanks for any reply :)

**Comment by evanbaldonado** [comment]
Hi! Try the following steps, and let me know if you run into any issues:

1. From your repository locally, run `bundle exec jekyll serve`.
2. Navigate to the generated _site folder, and identify the paths for any .html files that you want encrypted (PageCrypt can only encrypt .html files, but your .md files should be converted to .html files in _site).
3. Set up or modify your existing build/deployment workflow file (see the instructions on the readme for this repository if you haven't done this, yet). Depending on your configuration, this workflow will likely only run when actually deploying to GitHub pages and not when testing locally.
4. If your theme is set up differently, you might need to rewrite some of the code I have. The general idea is:
    - Install PageCrypt.
    - After the site has been built (but before it has been deployed), use PageCrypt to encrypt various files (in the _site folder). This repository shows how to have a list of files and passwords stored in a plaintext file and then iterate over that, but for your case it might be easier to just start by hard-coding in specific pagecrypt calls.

Also, note that this will only encrypt the resulting .html files on your deployed site and not any files (.html and .md and also the passwords file) in the repository itself. Given that your repository is currently public, people are still able to read your pages there. Hope this helps!

**Reply by ayan20985** [reply]
Many thanks and much love! Going to try this out later today.
